### PR TITLE
Added a number of photos to Land

### PR DESCRIPTION
**Pull request by tekker2234**
**Content (Artwork)**

This PR adds 33 new photos to the Land folder for people to use however they like. 

The pieces marked with the suffix -DAK are provided by me using the [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/) license.
The pieces marked with the suffix -Senn are provided by Senn on discord using the [CC0](https://creativecommons.org/public-domain/cc0/) license (in the public domain).

The asset PR is https://github.com/endless-sky/endless-sky-assets/pull/176
the high DPI PR is https://github.com/endless-sky/endless-sky-high-dpi/pull/419

### CODE REVIEW

**Comment by TheGiraffe3** [comment]
You’ll probably need to remove the endings and lowercase them. Would you like me to make a PR to your branch updating the copyright?

File: copyright [hunk 1]
```diff
@@ -1598,6 +1598,39 @@ Files:
...
+ images/land/water22*
+ images/land/water23*
+ images/land/water24*
+ images/land/water25*
+Copyright: DAK
+License: CC-BY-SA 4.0
+
+Files:
+ images/land/fields30*
+ images/land/mountain30*
```
Comment (TheGiraffe3) [inline copyright:1629]:
```suggestion
 images/land/mountain31*
```

File: copyright [hunk 2]
```diff
@@ -1598,6 +1598,39 @@ Files:
...
+ images/land/fields28*
+ images/land/fields29*
+ images/land/forest11*
+ images/land/forest12*
+ images/land/haze3*
+ images/land/path1*
+ images/land/path2*
+ images/land/path3*
+ images/land/path4*
+ images/land/sea23*
```
Comment (TheGiraffe3) [inline copyright:1616]:
```suggestion
 images/land/sea24*
```

**Comment by tekker2234** [comment]
> You’ll probably need to remove the endings and lowercase them. Would you like me to make a PR to your branch updating the copyright?

Thanks! Sorry for not responding earlier, I just woke up.

**Review by TheGiraffe3 — CHANGES_REQUESTED** [review]
~I know I added that~ but I hadn’t realized you would have to change the numbers as well.

**Comment by TheGiraffe3** [comment]
> > You’ll probably need to remove the endings and lowercase them. Would you like me to make a PR to your branch updating the copyright?
> 
> Thanks! Sorry for not responding earlier, I just woke up.

No problem.

**Comment by tekker2234** [comment]
> No problem. You can click the `Add to Batch` button and then `Commit Suggestions` to make my requested changes.

I clicked the button to commit suggestions without hitting add to batch. It seems I can't go back and redo it. Is that bad?

**Comment by TheGiraffe3** [comment]
No. It just makes commit history not quite as clean.

**Comment by tekker2234** [comment]
The copyright errors were that clouds9 and 10 appeared before clouds in the file and that the license that was specified was not valid. I tried to fix the license issue by copying the license line from another person in the doc that used a CC-BY-SA-4.0 license.

**Comment by TheGiraffe3** [comment]
Sorry for putting it in the wrong place. Any more errors, or does it work now?

**Comment by tekker2234** [comment]
I don't know how to initiate the check

**Comment by TheGiraffe3** [comment]
Go to the `Actions` tab in your repository and you should be able to click a button. I don’t remember what it says, but the one nearest the top should initiate the runs.

**Comment by tekker2234** [comment]
Screenshot 2024-06-23 105914
This is what I have

**Comment by TheGiraffe3** [comment]
It should look more like this:
IMG_2542

**Comment by tekker2234** [comment]
oh. I just pressed the big green button without reading it. should not have done that I guess.

**Comment by TheGiraffe3** [comment]
Big green button is the right one to do though…

**Comment by tekker2234** [comment]
it brought me to the screen I shared

File: copyright [hunk 3]
```diff
@@ -1598,6 +1598,37 @@ Files:
 Copyright: Michael Zahniser <mzahniser@gmail.com>
 License: CC-BY-SA-4.0
 
+Files:
```
Comment (TheGiraffe3) [inline copyright:1601]: You’ll have problems with the city files. It probably wouldn’t hurt to move it all close to the end of the file.

Comment (tekker2234) [reply copyright]: Just moved them to the end, after any other files in the same naming scheme

Comment (TheGiraffe3) [reply copyright]: Great, the workflows should be working now when they get run.

Comment (TheGiraffe3) [reply copyright]: (you can resolve this by pressing the `Resolve Conversation` button)

**Comment by TheGiraffe3** [comment]
Just realized that the `sea22` copyright is right underneath your copyright info and you’re adding `sea24`.

**Comment by tekker2234** [comment]
oh heck

**Comment by TheGiraffe3** [comment]
Well, that’s one way to do it. Looks good now, I think.

**Comment by roadrunner56** [comment]
Thanks for adding these! Most of these look good, although I have some issues with a couple of them:
clouds10: We generally want to avoid having examples of modern technology in landing images, which includes airplanes.
haze3 is a little too out of focus and blurry.
water26 is kind of washed out and hard to read.
Otherwise these look good in my opinion, and will go a long way towards helping with duplicate images (https://github.com/endless-sky/endless-sky/discussions/7107)

**Comment by tekker2234** [comment]
I'll also see if I can do anything with water26 to make it more legible, though no promises on that. I am pretty sure I would need to learn a lot more about Krita to fix it so I might not edit it if it takes too much time.

**Comment by tekker2234** [comment]
Yeah, idk what I'd do for it and I don't think I'll have too much time for at least the next week so I am not doing it for now. 
I'll be getting a new SD card reader soon and with that I'll be able to pull a bunch of photos off of my camera which I will either put into this PR or a new one if this gets merged. I'll take another crack at it then.

**Comment by TheGiraffe3** [comment]
> Thanks for adding these! Most of these look good, although I have some issues with a couple of them:
> haze3 is a little too out of focus and blurry.

I would think haze3 is fine. After all, it is a haze…

**Comment by nickshanks** [comment]
Some of these I don't believe add much to the game, namely: the three city pics (not 'alien'/futuristic enough), the two cloud ones (mundane, too Earth-like), and path3 is a nice bridge but seems out of place in a space game.

**Comment by TheGiraffe3** [comment]
Well, options are good. The three city pics, well, it’s the same with things such as `city26`.

**Comment by roadrunner56** [comment]
> > Thanks for adding these! Most of these look good, although I have some issues with a couple of them:
> > haze3 is a little too out of focus and blurry.
> 
> I would think haze3 is fine. After all, it is a haze…

If you look at other haze photos, they do have a lot of cloud cover yes, but the image itself is focused and sharp. That's ideally what we want to see for landing images.

**Comment by tekker2234** [comment]
> Some of these I don't believe add much to the game, namely: the three city pics (not 'alien'/futuristic enough), the two cloud ones (mundane, too Earth-like), and path3 is a nice bridge but seems out of place in a space game.
> Well, options are good. The three city pics, well, it’s the same with things such as `city26`. Clouds could be in a gas giant. The `path3` could be with the Xapleux, if I remember their shape correctly.

So should I keep the clouds photos and path 3 in or should I remove them as well?

**Comment by roadrunner56** [comment]
I think the city photos were fine. There are already plenty of landing images that are just photos of modern day cities. There are also mundane cloud images already used in the game, so I don't necessarily see that as a problem either. 
Lots of places in ES aren't your stereotypical shiny sci-fi future; there's a place for more naturalistic images like path3 and mundane cityscapes.

**Comment by tekker2234** [comment]
> I think the city photos were fine. There are already plenty of landing images that are just photos of modern day cities. There are also mundane cloud images already used in the game, so I don't necessarily see that as a problem either. Lots of places in ES aren't your stereotypical shiny sci-fi future; there's a place in this game for more naturalistic images like path3 and mundane cityscapes.

So should I add the three city photos back in? I don't know too much about where ES development is right now and what exactly y'all are looking for so I'll just wait a day or so before taking anything else out to let people reason it out.
For now I'll add the city photos back in so people can see them and have that inform their discussions.

**Review by TheGiraffe3 — APPROVED** [review]

**Review by roadrunner56 — APPROVED** [review]

**Comment by tekker2234** [comment]
Added a bunch of new photos that Senn gave me. I know the city photo is likely unwanted but I figured I'd put it in just in case.

File: copyright [hunk 4]
```diff
@@ -3466,6 +3466,47 @@ Comment:
...
+ images/land/city29*
+ images/land/fields30*
+ images/land/forest13*
+ images/land/forest14*
+ images/land/mountain31*
+ images/land/canyon16*
+ images/land/canyon17*
+ images/land/canyon18*
+ images/land/canyon19*
+ images/land/cave1*
```
Comment (TheGiraffe3) [inline copyright:3504]:
```suggestion
 images/land/forest13*
```

File: copyright [hunk 5]
```diff
@@ -3466,6 +3466,47 @@ Comment:
...
+ images/land/fog12*
```
Comment (TheGiraffe3) [inline copyright:3505]:
```suggestion
 images/land/forest14*
```

File: copyright [hunk 6]
```diff
@@ -3466,6 +3466,47 @@ Comment:
...
+ images/land/fog13*
```
Comment (TheGiraffe3) [inline copyright:3506]:
```suggestion
 images/land/mountain31*
```

**Review by TheGiraffe3 — CHANGES_REQUESTED** [review]
Alphabetical order. You might want to `Add to Batch` for these.

**Comment by tekker2234** [comment]
> Alphabetical order. You might want to `Add to Batch` for these.

sorry about that

**Review by TheGiraffe3 — APPROVED** [review]

**Review by beccabunny — CHANGES_REQUESTED** [review]
I would remove (or, if a large enough size is available to ignore any artifact, rotate and cut them down slightly) city28, city29, fields29, water22 and water23; the camera was leaning too much compared to the horizon, and in those in particular it seems to me too noticeable for vanilla. The rest is worthy in my opinion :thumbsup:

**Comment by tekker2234** [comment]
> I would remove (or, if a large enough size is available to ignore any artifact, rotate and cut them down slightly) city28, city29, fields29, water22 and water23; the camera was leaning too much compared to the horizon, and in those in particular it seems to me too noticeable for vanilla. The rest is worthy in my opinion 👍

I'll see what I can do

**Comment by tekker2234** [comment]
wait. what did I just do?!

**Comment by tekker2234** [comment]
I don't know what the merge commit that I just did was and the github change section is not being very helpful (to my inexperienced eye). Can anyone enlighten me?

**Comment by beccabunny** [comment]
I would do the same to city29

**Comment by Saugia** [comment]
We do use 2x's for all our landscape photos, however,

> I will upscale and provide 2x resolution versions of all the included photos sometime in the next few days.

Do *NOT* upscale photos. That degrades the quality of them and defeats the purpose of 2x sizes. The photos should be taken/originally uploaded at 2x size (1440 x 720), using that as the 2x images, and then in the base game, scaled down by half (to 720 x 360). I'm not sure how I feel about adding photos to the game if they don't have high dpi counterparts though, if that's not possible.

Also, some of these are, as 1x's, quite large. Most of our landscapes in the base game are ~ 40 to 100 KB. Several of these are breaking 400 KB. This is also kind of worked around with high dpi, as the purpose of having a high dpi plug-in is so that these sizes are reduced in the base game, and the quality and larger images can be grabbed as a plug-in for those interested.

**Comment by tekker2234** [comment]
> Do _NOT_ upscale photos. That degrades the quality of them and defeats the purpose of 2x sizes. The photos should be taken/originally uploaded at 2x size (1440 x 720), using that as the 2x images, and then in the base game, scaled down by half (to 720 x 360). I'm not sure how I feel about adding photos to the game if they don't have high dpi counterparts though, if that's not possible. 

Most of my photos were downscaled from ~3k resolution so I can (relatively) easily downscale and edit the originals to the 2x size. However the photos that Senn gave me are smaller than the 2x size (~1000-1500 x 602) so I would need to scale them up if I were to use them. I would like to keep those in if it is possible, if it is not then I will remove them.
(Senn's are: canyon16-19, cave1, city29, fields30, fog12-13, forest13-14, and mountain31)

> Also, some of these are, as 1x's, quite large. Most of our landscapes in the base game are ~ 40 to 100 KB. Several of these are breaking 400 KB. This is also the point of high dpi, as the purpose of having a high dpi plug-in is so that these sizes are reduced in the base game, and the quality and larger images can be grabbed as a plug-in for those interested.

The larger 1x's are the ones without any compression on them, I can resubmit them with the compression to get them close to that range. (The two path ones that are ~400KB are ones that I did a lot of editing on without any compression, compressing them should bring the filesize _way_ down).

Sorry for not being very active or following up with the promised 2x versions before. Life's gotten very busy and I haven't had much time to follow or work on this. I should have some time around next weekend. I'll see what I can do then.

**Comment by tekker2234** [comment]
> Put the 2x's in a PR to the high-dpi repository, which is [here](https://github.com/endless-sky/endless-sky-high-dpi). For the assets, put each landscape image in GIMP and save the file. Take all the GIMP files and PR them to the asset repository, which is [here](https://github.com/endless-sky/endless-sky-assets).

Should I also save the 2x's as .xcf?
Also is saving them as .xcf's supposed to quadruple the filesize compared to saving them as JPEGs?

**Comment by Saugia** [comment]
For a 2x PR, you need the 1440 x 720 images PR'd to that repo.

For an asset PR, you need to have the images in a gimp file at the 2x size PR'd to that repo.

If you had resolutions higher than 1440 x 720, then feel free to crop them until they are 1440 x 720 (not scale down directly to). If you have to scale it down to avoid cropping too much off, do it by even numbers. For example, if you have an image of 3000 x 2000, scaling that down directly to 1440 x 720 might stretch it, so you'd scale it down first by half to 1500 x 1000, and then instead of scale to 1440 x 720, crop it to that.

Keeping a list for myself and others of those without 2x's:

canyon16-19
cave1
city29
fields30
fog12-13
forest13-14
mountain31

I'll see if we can make exceptions.

**Comment by tekker2234** [comment]
> For a 2x PR, you need the 1440 x 720 images PR'd to that repo.
> 
> For an asset PR, you need to have the images in a gimp file at the 2x size PR'd to that repo.

I dont know if I fully understand what you mean by this. Are you saying that I should also make a PR for the main repository of the high dpi plugin with the 2x JPEGs and then upload the GIMP files to the main and 2x asset repositories? Or should I do something different?

> If you had resolutions higher than 1440 x 720, then feel free to crop them until they are 1440 x 720 (not scale down directly to). If you have to scale it down to avoid cropping too much off, do it by even numbers. For example, if you have an image of 3000 x 2000, scaling that down directly to 1440 x 720 might stretch it, so you'd scale it down first by half to 1500 x 1000, and then instead of scale to 1440 x 720, crop it to that. (I believe you could also keep scaling by small numbers but only if every time you do it, the numbers at the end of each dimension are even).

The images are mostly too large to crop down and still be good, (below is city 27 for reference). For the 2x I have been scaling the horizontal resolution to 1440 (ends up as 1440 by 960) then cropping the vertical axis to 720 and choosing the best slice of the image to display. 
Hilariously Cropped city27

> There isn't necessarily a strict size cap, though I was saying that the largest files here (in the base game, not the high dpi repo) are around 100KB, and you have several files here averaging around 400KB, which is quite a jump. Preferably that be cut down, yeah.

I'll probably get to those tomorrow, sorry again for the delay.

> Keeping a list for myself and others of those without 2x's:
> 
> canyon16-19 cave1 city29 fields30 fog12-13 forest13-14 mountain31
> 
> I'll see if we can make exceptions.

I will talk to Senn and see if they have higher resolution versions of the photos.

**Comment by Saugia** [comment]
> I dont know if I fully understand what you mean by this. Are you saying that I should also make a PR for the main repository of the high dpi plugin with the 2x JPEGs and then upload the GIMP files to the main and 2x asset repositories? Or should I do something different?

Yes, you need to PR to https://github.com/endless-sky/endless-sky-high-dpi with 2x images and https://github.com/endless-sky/endless-sky-assets with assets. These things are provided, along with a checklist, in the PR template, which is why we strongly recommend reading through those instead of deleting it and writing your own thing.

**Comment by tekker2234** [comment]
> These things are provided, along with a checklist, in the PR template when you first made this PR (which is why we strongly recommend reading through those instead of deleting it and writing your own description).

I apologize, I did not understand the importance of the checklist at the time. I'll make sure to follow it (and the template) for future PRs.

**Comment by Saugia** [comment]
No worries.

**Comment by tekker2234** [comment]
I guess it already put links up for them but,
The asset PR is [here](https://github.com/endless-sky/endless-sky-assets/pull/176)
the high DPI PR is [here](https://github.com/endless-sky/endless-sky-high-dpi/pull/419)

**Comment by Saugia** [comment]
After talking with the other devs and looking at what we have currently in the game, we don't really want to make exceptions for photos that don't have High DPI counterparts; it would be inconsistent resolution-wise, and these photos aren't already in-use as is, so it would be better to retain that choice.

(That said, there *are* a few that currently exist in game without 2x versions, and we're looking to eventually replace those.)

**Comment by tekker2234** [comment]
Sad to hear but understandable, I have removed the images.

**Pull request merged**